docs/apache-airflow/installation/installing-from-pypi.rst

```diff
@@ -79,53 +87,64 @@ where:
 - ``AIRFLOW_VERSION`` - Airflow version (e.g. :subst-code:`|version|`) or ``main``, ``2-0``, for latest development version
 - ``PYTHON_VERSION`` Python version e.g. ``3.8``, ``3.9``
 
-There is also a ``constraints-no-providers`` constraint file, which contains just constraints required to
-install Airflow core. This allows to install and upgrade airflow separately and independently from providers.
+However, that should not prevent you from being able to install providers and dependencies that were upgraded
+after the release to keep your installation up-to-dated, even if you do not upgrade airflow core version.
+Installing such dependencies should be done without constraints as a separate pip command.
 
-You can create the URL to the file substituting the variables in the template below.
+You should make sure to also add the ``apache-airflow`` package to the list of packages to install and pin
+it to the version that you have, otherwise you might end up with a different version of Airflow than you
+expect as ``pip`` can upgrade/downgrade it automatically when performing dependency resolution.
 
-.. code-block::
+.. code-block:: bash
 
-  https://raw.githubusercontent.com/apache/airflow/constraints-${AIRFLOW_VERSION}/constraints-no-providers-${PYTHON_VERSION}.txt
+    pip install "apache-airflow[celery]==|version|" --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-|version|/constraints-3.8.txt"
+    pip install "apache-airflow[celery]==|version|" apache-airflow-providers-google==10.1.1
 
-You can also use "latest" as version when you install "latest" stable version of Airflow. The "latest"
-constraints always points to the "latest" released Airflow version constraints:
+You can also downgrade providers or other dependencies this way if you need to (for example if you do not
+want to upgrade to the latest version of a provider).
 
-.. code-block::
+.. code-block:: bash
+
+    pip install "apache-airflow[celery]==|version|" --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-|version|/constraints-3.8.txt"
+    pip install "apache-airflow[celery]==|version|" dbt-core==0.20.0
+
+
+Not always such upgrade is possible - you might have conflicting dependencies, but by not using constraints,
```

Comment:
Such an upgrade is not always possible - you might have conflicting dependencies, but by not using constraints,
```